src/meta/src/hummock/manager/compaction_group_manager.rs

```diff
@@ -300,45 +299,50 @@ impl<S: MetaStore> HummockManager<S> {
             });
             modified_groups
                 .entry(group_id)
-                .and_modify(|count| *count -= 1)
-                .or_insert(
-                    current_version
-                        .get_compaction_group_levels(group_id)
-                        .member_table_ids
-                        .len() as u64
-                        - 1,
-                );
+                .or_insert_with(|| vec![])
+                .push(*table_id);
         }
 
         // Remove empty group, GC SSTs and remove metric.
         let mut branched_ssts = BTreeMapTransaction::new(&mut versioning.branched_ssts);
-        let groups_to_remove = modified_groups
-            .into_iter()
-            .filter_map(|(group_id, member_count)| {
-                if member_count == 0 && group_id > StaticCompactionGroupId::End as CompactionGroupId
+        let mut dirty_groups = vec![];
+        for (group_id, mut removed_table_ids) in modified_groups {
+            if let Some(group) = versioning.current_version.levels.get(&group_id) {
+                removed_table_ids.sort();
+                let mut dropped_sstables = vec![];
+                if group_id > StaticCompactionGroupId::End as CompactionGroupId
+                    && group.member_table_ids.eq(&removed_table_ids)
```

Comment:
member_table_ids is not guaranteed to be sorted, e.g. [here](https://github.com/risingwavelabs/risingwave/blob/a2ea8d4c80cb6bd3d65f91f0c0775011958fff10/src/storage/hummock_sdk/src/compaction_group/hummock_version_ext.rs#L456). Not sure if any other places.

Reply:
No. We sort it every time after we change it